pennylane/ops/functions/assert_valid.py

```diff
@@ -276,6 +295,37 @@ def _check_bind_new_parameters(op):
         assert qml.math.allclose(d1, d2), failure_comment
 
 
+def _check_differentiation(op):
+    """Checks that the operator can be executed and differentiated correctly."""
+
+    if op.num_params == 0:
+        return
+
+    data, struct = qml.pytrees.flatten(op)
+
+    def circuit(*args):
+        qml.apply(qml.pytrees.unflatten(args, struct))
+        return qml.probs(wires=op.wires)
+
+    qnode_ref = qml.QNode(circuit, qml.device("default.qubit"), diff_method="backprop")
+    qnode_ps = qml.QNode(circuit, qml.device("default.qubit"), diff_method="parameter-shift")
+
+    params = [x if isinstance(x, int) else qml.numpy.array(x) for x in data]
+
+    ps = qml.jacobian(qnode_ps)(*params)
+    expected_bp = qml.jacobian(qnode_ref)(*params)
+
+    if isinstance(ps, tuple):
+        for actual, expected in zip(ps, expected_bp):
+            assert qml.math.allclose(
+                actual, expected
+            ), "Parameter shift does not produce the expected Jacobian with this operator."
+    else:
+        assert qml.math.allclose(
+            ps, expected_bp
+        ), "Parameter shift does not produce the expected Jacobian with this operator."
```

Comment:
If in the future there's a case where the parameter-shift method provides the correct result (while backprop does not), should the error message be more conservative and report that there is simply a contradiction between the two results?

As you mention in the PR description, this already happened with `StatePrep` and for `QubitUnitary` (unless I am missing something)

Reply:
Makes sense, I'll update.